### PR TITLE
local dates

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?
Show times in user's timezones.

## Screenshots

| Before | After UK | After US Pacific | After US Eastern |
| --- | --- | --- | --- |
|  |  |  |  |

### CODE REVIEW

File: src/client/article.ts [hunk 1]
```diff
@@ -275,6 +275,20 @@ function hydrateQuizAtoms(): void {
...
 }
 
+function localDates(): void {
+    const date = document.querySelector("time.js-date");
+    const dateString = date?.getAttribute("data-date");
+    if (!dateString || !date) return;
+    try {
+        const localDate = new Date(dateString);
+        if (isValidDate(localDate)) {
+            date.innerHTML = formatLocal(localDate);
```
Comment (JamieB-gu) [inline src/client/article.ts:285]:
```suggestion
            date.textContent = formatLocal(localDate);
```

File: src/client/article.ts [hunk 2]
```diff
@@ -275,6 +275,20 @@ function hydrateQuizAtoms(): void {
...
+        }
+    } catch(e) {
+        console.error(e);
```
Comment (JamieB-gu) [inline src/client/article.ts:288]:
```suggestion
        console.error("Could not set a local date", e);
```

File: src/date.ts [hunk 3]
```diff
@@ -87,9 +87,21 @@ const padZero = (n: number): string =>
...
     `${padZero(date.getUTCHours())}.${padZero(date.getUTCMinutes())}`;
 
+const localTime = (date: Date): string =>
+    `${padZero(date.getHours())}.${padZero(date.getMinutes())}`;
+
 const format = (date: Date): string =>
     `${day(date)} ${date.getUTCDate()} ${month(date)} ${date.getUTCFullYear()} ${time(date)} UTC`;
 
+const formatLocal = (date: Date): string =>
+    `${localDay(date)} ${date.getDate()} ${localMonth(date)} ${date.getFullYear()} ${localTime(date)}`;
```
Comment (JamieB-gu) [inline src/date.ts:97]: Nice! Dotcom also includes the timezone at the end of this (e.g. BST, EDT) - do we want to do that too?

**Review by JamieB-gu — COMMENTED** [review]
Elegant implementation 👍

Couple of small suggestions.

File: src/date.ts (hunk 3 above)
Comment (webb04) [reply src/date.ts]: I think that would be a nice addition, I've pushed the following change. Couldn't see an easy way to display the shortened timezone
Screen Shot 2020-11-02 at 12 34 09

File: src/date.ts [hunk 4]
```diff
@@ -87,9 +87,24 @@ const padZero = (n: number): string =>
...
+    `${padZero(date.getHours())}.${padZero(date.getMinutes())}`;
+
+const localTimeZone = (date: Date): string =>
+    /\(.*?\)$/.exec(date.toTimeString())?.pop() ?? "";
+
 const format = (date: Date): string =>
     `${day(date)} ${date.getUTCDate()} ${month(date)} ${date.getUTCFullYear()} ${time(date)} UTC`;
 
+const formatLocal = (date: Date): string =>
+    `${localDay(date)} ${date.getDate()} ${localMonth(date)} ${date.getFullYear()} ${localTime(date)} ${localTimeZone(date)}`;
```
Comment (dskamiotis) [inline src/date.ts:100]: to me this seems very usable and understandable 👍

**Review by dskamiotis — APPROVED** [review]

**Review by dskamiotis — APPROVED** [review]

**Pull request merged**